mozci/console/commands/push.py

```diff
@@ -314,6 +335,75 @@ def _serialize_regressions(regressions):
                     emails, matrix_room, push, previous, classification, regressions
                 )
 
+    def retrigger_failures(self, groups, count, allowed_patterns):
+        groups_with_failures = {
+            name: failing_tasks
+            for name, failing_tasks in groups.items()
+            if failing_tasks
+        }
+        if not groups_with_failures:
+            return
+
+        # If there is more than one group with failing tasks, we should retrigger only one of them
```

Comment:
Sorry I realize this wasn't clear in https://github.com/mozilla/mozci/issues/654#issuecomment-1145889375.
We should retrigger all groups, and if there is more than one task for a given group we should retrigger only one of the tasks.